### ISSUE TITLE
"bus error" when running under docker

### ISSUE THREAD

**Issue by horacimacias** [tinyproxy/tinyproxy#249]
I've been using a docker container which uses tinyproxy without any problems for weeks.
Today, without any apparent reason, that same container image started failing to start tinyproxy.
Console output shows "Bus error".

Can somebody please suggest any way to fix or at least troubleshoot this?

I've checked if this is related to shared memory being full (https://goblincoding.com/2018/02/19/docker-bus-error-no-space-left-on-device/) but that is not the case.
df -h shows:

> Filesystem                Size      Used Available Use% Mounted on
> overlay                  58.4G     56.0G         0 100% /
> tmpfs                    64.0M         0     64.0M   0% /dev
> tmpfs                     1.9G         0      1.9G   0% /sys/fs/cgroup
> /dev/sda1                58.4G     56.0G         0 100% /etc/resolv.conf
> /dev/sda1                58.4G     56.0G         0 100% /etc/hostname
> /dev/sda1                58.4G     56.0G         0 100% /etc/hosts
> shm                      64.0M         0     64.0M   0% /dev/shm

I'm running the container with privileged=true in case that matters.
tinyproxy logs are empty :(
Container OS is alpine 3.6.2
tinyproxy 1.8.4

Any idea?

**Comment by horacimacias** [comment]
in case this is of any help, I was using:

> User tinyproxy
> Group tinyproxy

in the config.
I switched user to 'root' which is less than ideal, but at least with that it works fine.
At least it's a workaround but if anybody has any suggestions to troubleshoot further I'd be happy to hear.

**Comment by rofl0r** [comment]
> Any idea?

i guess you could run a more recent version for starters. 1.8.4 is like 4 years old.

**Comment by horacimacias** [comment]
thanks.
I tried with alpine 3.10 and tinyproxy 1.10 too and had the same "Bus error" result.

**Comment by rofl0r** [comment]
run it like this:
`gdb --args tinyproxy -d -c config.conf`

then type 'r' inside gdb for "run"
and when you get the SIGBUS, print a backtrace with `bt`.

**Comment by horacimacias** [comment]
thanks again.

I ran it but looks like I didn't get much information.
Here you can see it inside gdb and outside

> /tmp # gdb --args tinyproxy -d -c /etc/tinyproxy/tinyproxy.conf 
> GNU gdb (GDB) 7.12.1
> Copyright (C) 2017 Free Software Foundation, Inc.
> License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
> This is free software: you are free to change and redistribute it.
> There is NO WARRANTY, to the extent permitted by law.  Type "show copying"
> and "show warranty" for details.
> This GDB was configured as "x86_64-alpine-linux-musl".
> Type "show configuration" for configuration details.
> For bug reporting instructions, please see:
> <http://www.gnu.org/software/gdb/bugs/>.
> Find the GDB manual and other documentation resources online at:
> <http://www.gnu.org/software/gdb/documentation/>.
> For help, type "help".
> Type "apropos word" to search for commands related to "word"...
> Reading symbols from tinyproxy...(no debugging symbols found)...done.
> (gdb) r
> Starting program: /usr/sbin/tinyproxy -d -c /etc/tinyproxy/tinyproxy.conf
> warning: Error disabling address space randomization: Operation not permitted
> During startup program terminated with signal SIGBUS, Bus error.
> (gdb) bt
> No stack.
> (gdb) quit
> /tmp # tinyproxy -d -c /etc/tinyproxy/tinyproxy.conf 
> Bus error

**Comment by horacimacias** [comment]
...after running with privileged=true in docker:

Starting program: /usr/sbin/tinyproxy -d -c /etc/tinyproxy/tinyproxy.conf

> Program received signal SIGBUS, Bus error.
> 0x00007ffff7dc22e7 in memset () from /lib/ld-musl-x86_64.so.1
> (gdb) bt
> #0  0x00007ffff7dc22e7 in memset () from /lib/ld-musl-x86_64.so.1
> #1  0x0000555555559915 in ?? ()
> #2  0x0000555555557dd4 in ?? ()
> #3  0x0000555555556e52 in ?? ()
> #4  0x00007ffff7d8f964 in __libc_start_main () from /lib/ld-musl-x86_64.so.1
> #5  0x0000000000000000 in ?? ()

**Comment by horacimacias** [comment]
in case this is of any help, here's the full run with a "brand new" / clean alpine 3.6 container:

> docker run --rm -it --privileged=true alpine:3.6 sh                                                                                                                                                                    22:21:58
> / # apk update
> fetch http://dl-cdn.alpinelinux.org/alpine/v3.6/main/x86_64/APKINDEX.tar.gz
> fetch http://dl-cdn.alpinelinux.org/alpine/v3.6/community/x86_64/APKINDEX.tar.gz
> v3.6.5-44-gda55e27396 [http://dl-cdn.alpinelinux.org/alpine/v3.6/main]
> v3.6.5-34-gf0ba0b43d5 [http://dl-cdn.alpinelinux.org/alpine/v3.6/community]
> OK: 8445 distinct packages available
> / # apk add tinyproxy gdb
> (1/12) Installing expat (2.2.0-r1)
> (2/12) Installing ncurses-terminfo-base (6.0_p20171125-r1)
> (3/12) Installing ncurses-terminfo (6.0_p20171125-r1)
> (4/12) Installing ncurses-libs (6.0_p20171125-r1)
> (5/12) Installing libbz2 (1.0.6-r5)
> (6/12) Installing libffi (3.2.1-r3)
> (7/12) Installing gdbm (1.12-r0)
> (8/12) Installing readline (6.3.008-r5)
> (9/12) Installing sqlite-libs (3.25.3-r0)
> (10/12) Installing python2 (2.7.15-r0)
> (11/12) Installing gdb (7.12.1-r1)
> (12/12) Installing tinyproxy (1.8.4-r3)
> Executing tinyproxy-1.8.4-r3.pre-install
> Executing busybox-1.26.2-r11.trigger
> OK: 58 MiB in 25 packages
> / # tinyproxy -d
> Bus error
> / # gdb --args tinyproxy -d -c /etc/tinyproxy/tinyproxy.conf
> GNU gdb (GDB) 7.12.1
> Copyright (C) 2017 Free Software Foundation, Inc.
> License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
> This is free software: you are free to change and redistribute it.
> There is NO WARRANTY, to the extent permitted by law.  Type "show copying"
> and "show warranty" for details.
> This GDB was configured as "x86_64-alpine-linux-musl".
> Type "show configuration" for configuration details.
> For bug reporting instructions, please see:
> <http://www.gnu.org/software/gdb/bugs/>.
> Find the GDB manual and other documentation resources online at:
> <http://www.gnu.org/software/gdb/documentation/>.
> For help, type "help".
> Type "apropos word" to search for commands related to "word"...
> Reading symbols from tinyproxy...(no debugging symbols found)...done.
> (gdb) r
> Starting program: /usr/sbin/tinyproxy -d -c /etc/tinyproxy/tinyproxy.conf
> 
> Program received signal SIGBUS, Bus error.
> 0x00007ffff7dc22e7 in memset () from /lib/ld-musl-x86_64.so.1
> (gdb) bt
> #0  0x00007ffff7dc22e7 in memset () from /lib/ld-musl-x86_64.so.1
> #1  0x0000555555559915 in ?? ()
> #2  0x0000555555557dd4 in ?? ()
> #3  0x0000555555556e52 in ?? ()
> #4  0x00007ffff7d8f964 in __libc_start_main () from /lib/ld-musl-x86_64.so.1
> #5  0x0000000000000000 in ?? ()
> (gdb)

**Comment by horacimacias** [comment]
..the crazy thing is, running the same commands on another host running docker, I have no problems at all =:O

**Comment by horacimacias** [comment]
ok in case anybody else has a similar issue, this must be something on my docker setup as same image works fine on other docker hosts.

A couple weeks ago I was having issues because the Docker.qcow2 file was reaching the limit (I'm running docker on OSX). I stopped docker, renamed that file, started docker so a new Docker.qcow2 was created (loosing the previously stored images, beware).

At least for me, starting from a clean Docker.qcow2 file made the same image work (I had to rebuild it but it was the same Dockerfile). So, all in all, just a coincidence I was running tinyproxy; doesn't look like this is an issue on tinyproxy itself at all.

The location of the Docker.qcow2 file is ~/Library/Containers/com.docker.docker/Data/com.docker.driver.amd64-linux/Docker.qcow2 

@rofl0r thanks for answering and sorry for the confusion!
